outputs/email.go

```diff
@@ -13,10 +13,40 @@ import (
 	"github.com/aquasecurity/postee/v2/layout"
 )
 
+type EmailSender interface {
+	Send(address string, to []string, body []byte) error
+}
+
+type EmailConfig struct {
+	Username   string
+	Password   string
+	ServerHost string
+	ServerPort string
+	SenderAddr string
+}
+
 var (
 	errThereIsNoRecipient = errors.New("there is no recipient")
+	lookupMXFunc          = net.LookupMX
 )
 
+type emailSender struct {
+	conf EmailConfig
+	send func(string, smtp.Auth, string, []string, []byte) error
+}
+
+func (e *emailSender) Send(address string, to []string, body []byte) error {
+	var auth smtp.Auth
+	if len(e.conf.Password) > 0 && len(e.conf.Username) > 0 {
+		auth = smtp.PlainAuth("", e.conf.Username, e.conf.Password, e.conf.ServerHost)
+	}
+	return e.send(address, auth, e.conf.SenderAddr, to, body)
```

Comment:
@afdesk - this logic should handle the case where user passes empty user/pass and expects to not use an mx server.

Reply:
method surprisingly has a logic to handle authorization and this isn't reflected in method name. Not clear why it couldn't be part of output Send method. Affects readability as well

Reply:
I agree - this is a good point. I've simplified the code to just be able to override sendFunc without requiring additional interfaces.